From 97d7d69c4d466e97b080bbb2ff94365e5f5d0219 Mon Sep 17 00:00:00 2001
From: Ankit Jain <ankit-aj.jain@broadcom.com>
Date: Fri, 20 Jun 2025 11:33:58 +0000
Subject: [PATCH] llvm: CodeGen: Fix build failure for MI->dump()

- MI->dump() i.e. machine instruction dump
 should only be used when building with debug type
- With release build type, code shouldn't compile
 MI->dump().
- Since, printSpillReloadChain() is being called
 only from LLVM_DEBUG() which shouldn't get compiled
 for release build type.
 So, moved the definition of printSpillReloadChain()
 under suitable debug macro

Signed-off-by: Ankit Jain <ankit-aj.jain@broadcom.com>
---
 lib/CodeGen/MachineCopyPropagation.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/lib/CodeGen/MachineCopyPropagation.cpp b/lib/CodeGen/MachineCopyPropagation.cpp
index 9a0ab300b..15aff251f 100644
--- a/lib/CodeGen/MachineCopyPropagation.cpp
+++ b/lib/CodeGen/MachineCopyPropagation.cpp
@@ -1104,6 +1104,7 @@ void MachineCopyPropagation::BackwardCopyPropagateBlock(
   Tracker.clear();
 }
 
+#ifndef NDEBUG
 static void LLVM_ATTRIBUTE_UNUSED printSpillReloadChain(
     DenseMap<MachineInstr *, SmallVector<MachineInstr *>> &SpillChain,
     DenseMap<MachineInstr *, SmallVector<MachineInstr *>> &ReloadChain,
@@ -1115,6 +1116,7 @@ static void LLVM_ATTRIBUTE_UNUSED printSpillReloadChain(
   for (MachineInstr *MI : RC)
     MI->dump();
 }
+#endif
 
 // Remove spill-reload like copy chains. For example
 // r0 = COPY r1
-- 
2.40.4

